oxidation/crates/parsec_api_types/src/ext_types.rs

```diff
@@ -410,3 +410,86 @@ pub mod maybe_field {
 
     // serialize is not needed given we never omit fields
 }
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum Maybe<T> {
+    Present(T),
+    Absent,
+}
+
+impl<T> Default for Maybe<T> {
+    fn default() -> Self {
+        Self::Absent
+    }
+}
+
+impl<T> Maybe<T> {
+    pub fn is_absent(&self) -> bool {
+        matches!(self, Self::Absent)
+    }
+}
+
+impl<T, U> serde_with::SerializeAs<Maybe<T>> for Maybe<U>
+where
+    U: serde_with::SerializeAs<T>,
+{
+    fn serialize_as<S>(source: &Maybe<T>, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        match *source {
+            Maybe::Present(ref value) => {
+                serializer.serialize_some(&serde_with::ser::SerializeAsWrap::<T, U>::new(value))
+            }
+            Maybe::Absent => serializer.serialize_none(),
+        }
+    }
+}
+
+impl<'de, T, U> serde_with::DeserializeAs<'de, Maybe<T>> for Maybe<U>
+where
+    U: serde_with::DeserializeAs<'de, T>,
+    T: Default,
+{
+    fn deserialize_as<D>(deserializer: D) -> Result<Maybe<T>, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        struct MaybeVisitor<T, U>(std::marker::PhantomData<(T, U)>);
+
+        impl<'de, T, U> serde::de::Visitor<'de> for MaybeVisitor<T, U>
+        where
+            U: serde_with::DeserializeAs<'de, T>,
+            T: Default,
+        {
+            type Value = Maybe<T>;
+
+            fn expecting(&self, formatter: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+                formatter.write_str("maybe")
```

Comment:
According to documentation, expecting should be:

> The message should complete the sentence "This Visitor expects to receive ...", for example the message could be "an integer between 0 and 64". The message should not be capitalized and should not end with a period.

Now I don't see when this expecting would be used in our usecase (given field not present is ok and field present mean we pass it to T), so maybe have an unreachable!() instead ?